pkg/apply/taskrunner/context.go

```diff
@@ -38,20 +40,40 @@ func (tc *TaskContext) EventChannel() chan event.Event {
 // ResourceApplied updates the context with information about the
 // resource identified by the provided id. Currently, we keep information
 // about the generation of the resource after the apply operation completed.
-func (tc *TaskContext) ResourceApplied(id object.ObjMetadata, gen int64) {
+func (tc *TaskContext) ResourceApplied(id object.ObjMetadata, uid types.UID, gen int64) {
 	tc.appliedResources[id] = applyInfo{
 		generation: gen,
+		uid:        uid,
 	}
 }
 
+// ResourceUID looks up the UID of the given resource
+func (tc *TaskContext) ResourceUID(id object.ObjMetadata) (types.UID, bool) {
+	ai, found := tc.appliedResources[id]
+	if !found {
+		return "", false
+	}
+	return ai.uid, true
+}
+
+// AllResourceUIDs returns a set with the UIDs of all the resources in the
+// context.
+func (tc *TaskContext) AllResourceUIDs() sets.String {
+	uids := sets.NewString()
+	for _, ai := range tc.appliedResources {
+		uids.Insert(string(ai.uid))
+	}
+	return uids
+}
+
 // ResourceGeneration looks up the generation of the given resource
 // after it was applied.
-func (tc *TaskContext) ResourceGeneration(id object.ObjMetadata) int64 {
+func (tc *TaskContext) ResourceGeneration(id object.ObjMetadata) (int64, bool) {
 	ai, found := tc.appliedResources[id]
 	if !found {
-		return 0
+		return 0, false
```

Comment:
Doesn't 0 imply the bool is false? Do we need 2 outputs here ?

Reply:
Using two return values is idiomatic go instead of using magic return values. It also fits with the rest of the API of the TaskContext.